### PR TITLE
Refactor min/max value update in Parquet statistics

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

File: datafusion/core/src/datasource/file_format/parquet.rs [hunk 1]
```diff
@@ -304,154 +304,75 @@ fn summarize_min_max(
...
-                        }
-                    }
-                    return;
-                }
+        ParquetStatistics::Double(s)
+            if DataType::Float64 == *fields[i].data_type() && s.has_min_max_set() =>
+        {
+            if let Some(max_value) = &mut max_values[i] {
+                max_value
+                    .update_batch(&[Arc::new(Float64Array::from(vec![Some(*s.max())]))])
```
Comment (Jefffrey) [inline datafusion/core/src/datasource/file_format/parquet.rs:368]:
```suggestion
                    .update_batch(&[Arc::new(Float64Array::from(vec![*s.max()]))])
```

I think this also works? Goes for the other branches too

File: datafusion/core/src/datasource/file_format/parquet.rs [hunk 2]
```diff
@@ -304,154 +304,75 @@ fn summarize_min_max(
...
-                            Ok(_) => {}
-                            Err(_) => {
-                                min_values[i] = None;
-                            }
-                        }
-                    }
-                    return;
-                }
+        ParquetStatistics::Boolean(s)
+            if DataType::Boolean == *fields[i].data_type() && s.has_min_max_set() =>
```
Comment (Jefffrey) [inline datafusion/core/src/datasource/file_format/parquet.rs:308]: Is it possible to pull out the `s.has_min_max_set()` check from the branches to a once off check at the start of the function? I think `ParquetStatistics` has an equivalent `has_min_max_set()` :thinking:

**Review by Jefffrey — APPROVED** [review]
Nice refactoring :+1: 

Left some tips to see if it's possible to simplify it even further

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
That is a really nice cleanup -- thank you @Weijun-H  and @Jefffrey

**Pull request merged**